full/src/main/java/apoc/uuid/Uuid.java

```diff
@@ -37,12 +37,20 @@ public Stream<UuidInstallInfo> install(@Name("label") String label, @Name(value
         final String uuidFunctionName = getUuidFunctionName();
         Map<String, Object> addToExistingNodesResult = Collections.emptyMap();
         if (uuidConfig.isAddToExistingNodes()) {
-            addToExistingNodesResult = Util.inTx(db, pools, txInThread ->
+            try {
+                addToExistingNodesResult = Util.inTx(db, pools, txInThread ->
                     txInThread.execute("CALL apoc.periodic.iterate(" +
                             "\"MATCH (n:" + Util.sanitizeAndQuote(label) + ") RETURN n\",\n" +
                             "\"SET n." + Util.sanitizeAndQuote(uuidConfig.getUuidProperty()) + " = " + uuidFunctionName + "()\", {batchSize:10000, parallel:true})")
                             .next()
-            );
+                );
+            } catch (RuntimeException e) {
+                if (e.getMessage().contains( "There is no procedure with the name `apoc.periodic.iterate` registered for this database instance" )) {
+                    throw new RuntimeException("apoc core needs to be installed when using apoc.uuid.install with the flag addToExistingNodes = true");
+                } else {
+                    throw e;
+                }
+            }
```

Comment:
Good catch.

I think this is fine for now, but it might be nice to think about how we should handle this type of stuff in the long term.